Jetcaster/app/src/main/java/com/example/jetcaster/ui/home/HomeViewModel.kt

```diff
@@ -129,23 +75,29 @@ class HomeViewModel(
                 selectedHomeCategory,
                 podcastStore.followedPodcastsSortedByLastEpisode(limit = 20),
                 refreshing,
-                discover,
-                podcastCategory,
-                libraryEpisodes
+                _selectedCategory.flatMapLatest { selectedCategory ->
+                    filterableCategoriesUseCase(selectedCategory).onEach {
+                        _selectedCategory.value = it.selectedCategory
```

Comment:
I'm a bit afraid of this, because if this wasn't a `StateFlow`, which automatically uses `distinctUntilChanged`, then this would cause infinite loop.  I think we're mixing kind of two states `_selectedCategory` and `HomeViewState` in one viewmodel. Anyway, I think it can be beyond this PR to change that.

What I would change for readability though is to just have as part of the combine lambda, so you don't have to have tne `onEach` because that what happens anyway:
```

_selectedCategory.value = filterableCategories.selectedCategory

HomeViewState(
...
```

Reply:
The original intent of this PR is to simplify the responsibility of `HomeViewModel` without changing the implementation (so mostly refactoring and moving code and not changing the underlying implementation). I agree that the current implementation can be improved though so let's address that separately.